sherpa/stats/__init__.py

```diff
@@ -147,7 +148,8 @@ def __init__(self, name='cash'):
         Likelihood.__init__(self, name)
 
     @staticmethod
-    def calc_stat(data, model, staterror=None, syserror=None, weight=None):
+    def calc_stat(data, model, staterror=None, syserror=None, weight=None, \
+                      bkg=None):
```

Comment:
As suggested by Omar, I'm currently updating the user stat class so as
to be backwards compatible.  The new calc_stat will have the following
signature

def calc_stat(data, model, staterror=None, syserror=None, weight=None,
**kwargs):

The UserStat class may or may not have the bkg, see backwards
compatibility comment above.  So I don't think we should raise an
exception if there is not bkg data.  For the wstat, an exception shall
be raised since as you remember from the conversation with K. Arnaud
that the bkg data must be used


On Wed, Aug 26, 2015 at 10:18 AM, Doug Burke <notifications@github.com>
wrote:

> In sherpa/stats/__init__.py
> <https://github.com/sherpa/sherpa/pull/90#discussion_r37986821>:
>
> > @@ -147,7 +148,8 @@ def __init__(self, name='cash'):
> >          Likelihood.__init__(self, name)
> >
> >      @staticmethod
> > -    def calc_stat(data, model, staterror=None, syserror=None, weight=None):
> > +    def calc_stat(data, model, staterror=None, syserror=None, weight=None, \
> > +                      bkg=None):
>
> Should we verify that bkg is None here (to catch the case when a user is
> calling something incorrectly)? For instance, add
>
> if bkg is None:
>     raise ValueError("The background can not be included in the statistic")
>
> but with an improved error message?
>
> Or is this limiting, and we may actually get cases where bkg is set here
> but its safe to ignore?
>
> This holds for all the remaining stats which have a bkg parameter but do
> not use it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sherpa/sherpa/pull/90/files#r37986821>.
>